src/lib/entities/audio/model/listening-history.ts

```diff
@@ -0,0 +1,20 @@
+import { localStorageAdapter, persistentWritable } from 'svelte-persistent-writable';
+
+/**
+ * The storage wih array of audio URLs, persists in localStorage
+ */
+export const listeningHistory = persistentWritable<number[]>([], {
+  storage: localStorageAdapter('listening-history'),
+});
+
+/**
+ * Function that adds a podcast id to the beginning of listening history.
+ * It removes any previous occurrences of the same podcast.
+ *
+ * @param podcastId {string} – an URl to add to history
+ */
+export const addToListeningHistory = (podcastId: number) => {
+  listeningHistory.update((history) => {
+    return [podcastId, ...history.filter((id) => id === podcastId)].slice(0, 6);
```

Comment:
question: Have you confirmed this works correctly? Looks like it should be a `!==` here

I'd appreciate if you added unit tests to your code as you write it, it's going to be quite a chore to add them later on